src/extension/user_tracking.c

```diff
@@ -0,0 +1,90 @@
+// Unless explicitly stated otherwise all files in this repository are
+// dual-licensed under the Apache-2.0 License or BSD-3-Clause License.
+//
+// This product includes software developed at Datadog
+// (https://www.datadoghq.com/). Copyright 2021 Datadog, Inc.
+
+#include "user_tracking.h"
+#include "commands/request_exec.h"
+#include "helper_process.h"
+#include "logging.h"
+#include "php_compat.h"
+#include "src/extension/request_abort.h"
+#include "string_helpers.h"
+#include <zend_hash.h>
+
+static void (*_ddtrace_set_user)(INTERNAL_FUNCTION_PARAMETERS) = NULL;
+
+static PHP_FUNCTION(set_user_wrapper)
+{
+    zend_string *user_id;
+    HashTable *metadata = NULL;
+    zend_bool propagate = false;
+    if (zend_parse_parameters(ZEND_NUM_ARGS(), "S|hb", &user_id, &metadata,
+            &propagate) == SUCCESS) {
+        if (user_id != NULL) {
+            dd_find_and_apply_verdict_for_user(user_id);
+        }
+    }
+
+    // This shouldn't be necessary, if it is we have a bug
+    if (_ddtrace_set_user == NULL) {
```

Comment:
Since you're coping with it, I would add a phpt test without the `ddtrace`

Reply:
What exactly would you want the test to do?  Just check the log statement? We can't really test `set_user` without DDTrace...

Reply:
Check it does not end on a segmentation fault

Reply:
`set_user_wrapper` isn't exposed to the user so we can only call it indirectly as a result of calling `set_user` when available.

Reply:
You can call set user without the trace extension but it's ok